agents-docs/content/docs/typescript-sdk/memory.mdx

```diff
@@ -85,3 +85,55 @@ The tool results include both the input parameters and output results, formatted
   "results": [...]
 }
 ```
+
+## Conversation Compacting System
+
+For very long conversations that exceed model context limits, the framework includes an intelligent compacting system that automatically manages memory by condensing older messages while preserving essential context.
+
+### How Compacting Works
+
+The compacting system activates automatically when conversations approach token limits:
+
+<Steps>
+  <Step>
+    **Context Monitoring**: System continuously monitors conversation size against model limits
+  </Step>
+  <Step>
+    **Automatic Triggering**: Compacting triggers at 50% of context window for conversation-level, or at model-aware thresholds (~75-91% depending on model size) for sub-agent generation
+  </Step>
+  <Step>
+    **Tool Result Archiving**: Large tool results are stored as artifacts and replaced with summary references
+  </Step>
+  <Step>
+    **AI Summarization**: Older conversation parts are summarized by AI while preserving key context
+  </Step>
+  <Step>
+    **Fallback Protection**: If AI summarization fails, system falls back to simple message truncation
+  </Step>
+</Steps>
+
+### Model-Specific Behavior
+
+Different models have different context windows, and compacting adapts accordingly:
+
+| Model | Context Window | Conversation Threshold | Sub-Agent Generation Threshold |
+|-------|----------------|------------------------|--------------------------------|
+| GPT-5.1 | 400K tokens | 200K (50%) | ~332K (83%) |
+| Claude-4.5 Sonnet | 200K tokens | 100K (50%) | ~166K (83%) |
+| Gemini 3 Flash | 1M tokens | 500K (50%) | ~910K (91%) |
+
+### Compacting Types
+
+#### Conversation-Level Compacting
+- **Trigger**: When conversation reaches 50% of model's context window
+- **Action**: Compacts entire conversation history into summary + artifacts
+- **Use Case**: Long conversations with extensive history
+
+#### Sub-Agent Generation Compacting  
```

Comment:
It's not clear to me what the difference is between these compacting types. Can you provide examples in the docs to illustrate what the difference is between these?